server/src/lib.rs

```diff
@@ -110,29 +115,65 @@ pub struct AuthenticationResult {
     token: Option<String>,
 }
 
+#[derive(Debug,Clone)]
+enum AuthenticationInfos {
+    Kerberos(Arc<Mutex<kerberos::State>>),
+    Tls(String, String), // subject, thumbprint
+}
+
+#[derive(Debug)]
+pub enum AuthenticationPayload {
```

Comment:
I'm not sure that you need `AuthenticationPayload` as this is only a subset of `AuthenticationInfos`.